pkg/kubectl/cmd/util/factory.go

```diff
@@ -254,8 +254,8 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return kubectl.ReaperFor(mapping.Kind, client, expClient)
 		},
-		Validator: func() (validation.Schema, error) {
-			if flags.Lookup("validate").Value.String() == "true" {
+		Validator: func(validate bool) (validation.Schema, error) {
```

Comment:
I see. The caller should only call when they should validate, and we don't need the `validation.NullSchema{}` in Validator.